flight/Modules/Actuator/actuator.c

```diff
@@ -297,126 +277,36 @@ static void actuatorTask(void* parameters)
-				curve2 = CollectiveCurve(desired.Throttle, mixerSettings.ThrottleCurve2, MIXERSETTINGS_THROTTLECURVE2_NUMELEM);
-				break;
-			case MIXERSETTINGS_CURVE2SOURCE_ROLL:
-				curve2 = CollectiveCurve(desired.Roll, mixerSettings.ThrottleCurve2, MIXERSETTINGS_THROTTLECURVE2_NUMELEM);
-				break;
-			case MIXERSETTINGS_CURVE2SOURCE_PITCH:
-				curve2 = CollectiveCurve(desired.Pitch, mixerSettings.ThrottleCurve2, MIXERSETTINGS_THROTTLECURVE2_NUMELEM);
-				break;
-			case MIXERSETTINGS_CURVE2SOURCE_YAW:
-				curve2 = CollectiveCurve(desired.Yaw, mixerSettings.ThrottleCurve2, MIXERSETTINGS_THROTTLECURVE2_NUMELEM);
-				break;
-			case MIXERSETTINGS_CURVE2SOURCE_COLLECTIVE:
-				ManualControlCommandCollectiveGet(&curve2);
-				curve2 = CollectiveCurve(curve2, mixerSettings.ThrottleCurve2, MIXERSETTINGS_THROTTLECURVE2_NUMELEM);
-				break;
-			case MIXERSETTINGS_CURVE2SOURCE_ACCESSORY0:
-			case MIXERSETTINGS_CURVE2SOURCE_ACCESSORY1:
-			case MIXERSETTINGS_CURVE2SOURCE_ACCESSORY2:
-			case MIXERSETTINGS_CURVE2SOURCE_ACCESSORY3:
-			case MIXERSETTINGS_CURVE2SOURCE_ACCESSORY4:
-			case MIXERSETTINGS_CURVE2SOURCE_ACCESSORY5:
-				if(AccessoryDesiredInstGet(mixerSettings.Curve2Source - MIXERSETTINGS_CURVE2SOURCE_ACCESSORY0,&accessory) == 0)
-					curve2 = CollectiveCurve(accessory.AccessoryVal, mixerSettings.ThrottleCurve2, MIXERSETTINGS_THROTTLECURVE2_NUMELEM);
-				else
-					curve2 = 0;
-				break;
-		}
+		float curve2 = CollectiveCurve(
+			GetCurve2Source(&desired, mixerSettings.Curve2Source),
+			mixerSettings.ThrottleCurve2,
+			MIXERSETTINGS_THROTTLECURVE2_NUMELEM);
+
 
 		float * status = (float *)&mixerStatus; //access status objects as an array of floats
 
 		for(int ct=0; ct < MAX_MIX_ACTUATORS; ct++)
 		{
-			if(mixers[ct].type == MIXERSETTINGS_MIXER1TYPE_DISABLED) {
-				// Set to minimum if disabled.  This is not the same as saying PWM pulse = 0 us
-				status[ct] = -1;
-				command.Channel[ct] = 0.0f;
-				continue;
-			}
-
-			if((mixers[ct].type == MIXERSETTINGS_MIXER1TYPE_MOTOR) || (mixers[ct].type == MIXERSETTINGS_MIXER1TYPE_SERVO)) {
-				status[ct] = ProcessMixer(ct, curve1, curve2, &mixerSettings, &desired, dT);
-
-				// If a motor type, additional work to be done
-				if(mixers[ct].type == MIXERSETTINGS_MIXER1TYPE_MOTOR) {
-					if (status[ct] > 0) {
-						// Apply curve fitting, mapping the input to the propeller output.
-						status[ct] = actuatorSettings.MotorInputOutputCurveFit[ACTUATORSETTINGS_MOTORINPUTOUTPUTCURVEFIT_A] *
-						      powf(status[ct], actuatorSettings.MotorInputOutputCurveFit[ACTUATORSETTINGS_MOTORINPUTOUTPUTCURVEFIT_B]);
-					} else {
-						 // Idle throttle
-						status[ct] = 0.0f;
-					}
-				}
-			} else {
-				status[ct] = -1;
-			}
+			status[ct] = MixChannel(ct, &desired, curve1, curve2);
 
 			// Motors have additional protection for when to be on
-			if(mixers[ct].type == MIXERSETTINGS_MIXER1TYPE_MOTOR) {
+			if(GetMixerType(ct) == MIXERSETTINGS_MIXER1TYPE_MOTOR) {
```

Comment:
Not really.. but it's fairly cheap.  This rips out the whole type caching mechanism because looking up the type is fairly cheap (and if it's a performance problem GetMixerType can be made even cheaper).

That big long switch becomes:

```
  58 0000 0928                  cmp     r0, #9
  59 0002 2ED8                  bhi     .L5
  60 0004 DFE800F0              tbb     [pc, r0]
  61                    .L7:
  62 0008 05                    .byte   (.L6-.L7)/2
  63 0009 09                    .byte   (.L8-.L7)/2
  64 000a 0D                    .byte   (.L9-.L7)/2
  65 000b 11                    .byte   (.L10-.L7)/2
  66 000c 15                    .byte   (.L11-.L7)/2
  67 000d 19                    .byte   (.L12-.L7)/2
  68 000e 1D                    .byte   (.L13-.L7)/2
  69 000f 21                    .byte   (.L14-.L7)/2
  70 0010 25                    .byte   (.L15-.L7)/2
  71 0011 29                    .byte   (.L16-.L7)/2
```

TBB implements a jump table that takes 3-5 cycles.

It could be made cheaper at the expense of legibility with pointer arithmetic and DONT_BUILD_IF type asserts (it'd inline then).